src/pudl/analysis/demand_mapping.py

```diff
@@ -1,294 +1,942 @@
+        # Unzip because we can't use zipfile paths with geopandas
+        with zipfile.ZipFile(hifld_pa_zipfile, 'r') as zip_ref:
+            zip_ref.extractall(hifld_dir)
+            # Grab the UUID based directory name so we can change it:
+            extract_root = hifld_dir / \
+                pathlib.Path(zip_ref.filelist[0].filename).parent
+        extract_root.rename(hifld_pa_gdb_dir)
+    else:
+        logger.info("We've already got the planning area GeoDB.")
+
+    gdf = (
+        geopandas.read_file(hifld_pa_gdb_dir)
+        .assign(
+            SOURCEDATE=lambda x: pd.to_datetime(x.SOURCEDATE),
+            VAL_DATE=lambda x: pd.to_datetime(x.VAL_DATE),
+            ID=lambda x: pd.to_numeric(x.ID),
+            NAICS_CODE=lambda x: pd.to_numeric(x.NAICS_CODE),
+            YEAR=lambda x: pd.to_numeric(x.YEAR),
+        )
+        # Hack to work around geopanda issue fixed as of v0.8.0
+        # https://github.com/geopandas/geopandas/issues/1366
+        .assign(
+            ID=lambda x: x.ID.astype(pd.Int64Dtype()),
+            NAME=lambda x: x.NAME.astype(pd.StringDtype()),
+            COUNTRY=lambda x: x.COUNTRY.astype(pd.StringDtype()),
+            NAICS_CODE=lambda x: x.NAICS_CODE.astype(pd.Int64Dtype()),
+            NAICS_DESC=lambda x: x.NAICS_DESC.astype(pd.StringDtype()),
+            SOURCE=lambda x: x.SOURCE.astype(pd.StringDtype()),
+            VAL_METHOD=lambda x: x.VAL_METHOD.astype(pd.StringDtype()),
+            WEBSITE=lambda x: x.WEBSITE.astype(pd.StringDtype()),
+            ABBRV=lambda x: x.ABBRV.astype(pd.StringDtype()),
+            YEAR=lambda x: x.YEAR.astype(pd.Int64Dtype()),
+            PEAK_LOAD=lambda x: x.PEAK_LOAD.astype(float),
+            PEAK_RANGE=lambda x: x.PEAK_RANGE.astype(float),
+            SHAPE_Length=lambda x: x.SHAPE_Length.astype(float),
+            SHAPE_Area=lambda x: x.SHAPE_Area.astype(float),
+        )
     )
-    new_df["area_derived"] = new_df["geometry"].area
+    return gdf
+
+
+################################################################################
+# Demand allocation functions
+################################################################################
+def edit_id_set(row, new_ID, ID):
+    """
+    Editing ID sets by adding the new geometry ID if required.
+
+    This function edits original ID sets by adding the new geometry ID if
+    required. This function is called by another function
+    `complete_disjoint_geoms`
+    """
+    if row["geom_type"] == "geometry_new_int":
+        return frozenset(list(row[ID]) + [new_ID])
+
+    else:
+        return row[ID]
+
+
+def polygonize_geom(geom):
+    """
+    Remove zero-area geometries from a geometry collection.
+
+    Strip zero-area geometries from a geometrical object.
+    (maybe a single geometry object or collection e.g. GeometryCollection)
+    This function is called by another function `complete_disjoint_geoms`.
+    """
+    if type(geom) == GeometryCollection:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5664147)